R/zchunk_LB123.Electricity_china.R

```diff
@@ -0,0 +1,262 @@
+             coal = coal + coal.adj,
+             wind = wind + wind.adj,
+             nuclear = nuclear + nuc.adj,
+             solar = solar + solar.adj) %>%
+      select(-other, -row, -coal.adj, -wind.adj, -nuc.adj, -solar.adj) %>%
+      mutate(biomass = coal) %>%
+      gather(fuel, value, -province, -year) %>%
+      filter(value == 0 | value > 0.1) %>%
+      complete(nesting(province, fuel), year = HISTORICAL_YEARS_temp) %>%
+      arrange(province, fuel, year) %>%
+      group_by(province, fuel) %>%
+      mutate(value = approx_fun(year, value, rule = 2)) %>%
+      ungroup %>%
+      mutate(fuel = as.character(fuel)) ->
+      L132.GWh_province_F_elec_out
+
+    # Oil and gas does not exist in the CPSY so we will use the CESY for those
+    # Note these are in terms of energy in.  Since we are assuming constant efficiencies for all provinces
+    # mixing these with the elec_out will be fine for approtioning to provinces
+    L101.inNBS_Mtce_province_S_F %>%
+      filter(sector == "electricity", fuel %in% c("refined liquids", "gas")) %>%
+      replace_na(list(value = 0))  %>%
+      select(-sector) %>%
+      bind_rows(L132.GWh_province_F_elec_out)  %>%
+      # In Tibet, refined liquids inherits oil and gas inherits natural gas
+      mutate(fuel = replace(fuel, fuel == "oil" & province == "XZ", "refined liquids"),
+             fuel = replace(fuel, fuel == "natural.gas" & province == "XZ", "gas"),
+             # In Tibet our oil and gas numbers are in electricity out and they need to be energy in in Mtce. This is a total hack but
+             # just divide by the efficiency. Gas is 0.55 and oil is about 0.35
+             value =  replace(value, province == "XZ" & fuel %in% c("refined liquids", "gas"), 0),
+             copy_rl = value * 100 * CONV_GWH_EJ * CONV_EJ_MTCE / -0.35,
+             copy_g = value * 100 * CONV_GWH_EJ * CONV_EJ_MTCE / -0.55,
+             copy_rl = replace(value, province == "XZ" & fuel == "refined liquids", 0),
+             copy_g = replace(value, province == "XZ" & fuel == "gas", 0),
+             value = value + copy_rl + copy_g) %>%
+      select(-copy_g, -copy_rl) %>%
+      # Aggregate by fuel compute each province's percentage, by fuel
+      group_by(fuel, year) %>%
+      mutate(pct = value / sum(value)) %>%
+      ungroup %>%
+      mutate(fuel = replace(fuel, fuel == 'solar', 'solar PV')) %>%
+      select(-value) %>%
+      replace_na(list(pct = 0)) ->
+      L123.pct_province_elec_F
+
+    # Electricity generation inputs by fuel and province
+    # Note that fuel inputs are only available for selected fuels; province_share_data uses this relevant subset
+    L123.in_EJ_R_elec_F_Yh %>%
+      filter(GCAM_region_ID == gcamchina.REGION_ID, fuel %in% L123.pct_province_elec_F$fuel) ->
+      L123.in_EJ_province_elec_F
+    Fuel_in_Yh = unique(L123.in_EJ_R_elec_F_Yh$fuel)
```

Comment:
Don't need to create this list.
You can simply do `filter(fuel %in% unique(L123.in_EJ_R_elec_F_Yh$fuel), year %in% HISTORICAL_YEARS) %>%` at line 113